### DISCUSSION TITLE
Configurable info bars

### DISCUSSION THREAD

**Discussion by iarxm** [jarun/nnn#1685] [Show and tell]
Maybe this is a useful feature or already explored, but I've rendered the contexts and path within the statusbar.

I find I barely need to look at them and prefer an ultra minimal interface with nothing at the top.

Within the design principles, I believe this has a greater adherence to principles 2 and 3.

To implement this: 
- Code related to drawing the contexts, from 'redraw', was separated into it's own function. 
- This was also done for drawing the path, as a separte function. 
- Both those functions are then called from within 'statusbar' to include them within the statusbar.

I supposed that as functions, they could be more easily configured to be rendered at the top or bottom of the screen.

Then the file list was modified to begin on line 2 rather than 3. To do this:
- Within 'redraw': the loop for printing entries had to be modified to increment j at the end rather than beginning.
- Within 'drawline': a new variable cur_zero was introduced, to start the move from the 2nd xline rather than the 3rd. 'j' in redraw is then set to start at 'cur_zero' rather than 1. 
- For my implementation, i set cur_zero to 0, which provides a nice amount of padding from the screen top.

[cbar](https://github.com/futuresurf/nnn)

![scr23 07 07 t18 16 34](https://github.com/jarun/nnn/assets/101668581/3714480c-b4b8-4d7b-9818-53e30f1de5ee)

**Comment by jarun** [comment]
The context + path on the top and file information at the bottom provides a very clear separation. Adding all information at the bottom would lead to a clutter.
Also, in the image I don't see the information of the hovered entry. We can't lose that kind of information.

**Comment by iarxm** [comment]
I neglected to include that I replaced elements of the statusbar info with the context info to tackle the clutter problem

What you say is true if your project goal is to keep all the information always visible for the hovered file. The mod date and perms were removed in my fork, choosing the context info as higher priority.

For me personally, pressing 'd' is adequate to quickly display extra info, and correlates to about the same speed it takes for my eye to glimse downwards to the bottom corner.
My intention is to continue my fork to auto hide the status info and to bring it on top via a keypress. For me, there's no downtime pressing a key like 'd' whilst simultaneously glimpsing down.

Obviously there would be trade offs to this, and to integrate it into the mainline of the project would entail adding more configuration options which may not be desirable.

It won't be to everyones taste, but I thought I would share it incase anyone had interest. If there was interest could even add it as a patch.

I'm getting great use out of 'nnn', thanks for the great work!

**Reply by jarun** [reply]
> What you say is true if your project goal is to keep all the information always visible for the hovered file.

That's correct. I intend to keep the information shown consistently.

**Comment by iarxm** [comment]
Just on reflection, in a more general sense the porposition of my fork is: 'Configurable Information Bars'.

And if it were something anyone had interest in, then even adopting the function decomposition of the status bar and redraw it, it would allow greater readability of the code and would allow people to easily make their own mods if so desired

**Comment by jarun** [comment]
Closing this since fork exists and there are no plans to implement this in `nnn`.
